compiler+runtime/test/cpp/jank/runtime/obj/ratio.cpp

```diff
@@ -0,0 +1,401 @@
+#include "jank/runtime/obj/ratio.hpp"
+#include <doctest/doctest.h>
+#include <stdexcept>
+
+namespace jank::runtime
+{
+  TEST_SUITE("ratio_data")
+  {
+    TEST_CASE("Ratio Constructor")
+    {
+      SUBCASE("Valid Ratio")
+      {
+        obj::ratio_data ratio{ 4, 6 };
+        CHECK(ratio.numerator == 2); // Simplified
+        CHECK(ratio.denominator == 3);
```

Comment:
I think we want `CHECK_EQ` for all of these equality checks. That will allow doctest to provide better error messages when they're not equal.

Reply:
updated all. using `CHECK_EQ`, `CHECK_LT`, etc., instead.  for certain cases, this does not seem to work once `ratio_data` is not a jank object.

Reply:
> updated all. using CHECK_EQ, CHECK_LT, etc., instead. for certain cases, this does not seem to work once ratio_data is not a jank object.

this turns out to be caused by operators not being implemented in the `jank::runtime::obj` (but `ratio_data` is). everything works once operators are also moved into `jank::runtime::obj`.